src/bert/utils.py

```diff
@@ -0,0 +1,102 @@
+import logging as log
+
+from typing import Dict
+
+import torch
+import torch.nn as nn
+
+# huggingface implementation of BERT
+import pytorch_pretrained_bert
+
+class BertEmbedderModule(nn.Module):
+    """ Wrapper for BERT module to fit into jiant APIs. """
+
+    def __init__(self, args, cache_dir=None):
+        super(BertEmbedderModule, self).__init__()
+
+        self.model = \
+            pytorch_pretrained_bert.BertModel.from_pretrained(
+                args.bert_model_name,
+                cache_dir=cache_dir)
+        self.embeddings_mode = args.bert_embeddings_mode
+        # BERT supports up to 512 tokens; see section 3.2 of https://arxiv.org/pdf/1810.04805.pdf
+        assert args.max_seq_len <= 512
+        self.seq_len = args.max_seq_len
+
+        # Set trainability of this module.
+        for param in self.model.parameters():
+            param.requires_grad = bool(args.bert_fine_tune)
+
+    def forward(self, sent: Dict[str, torch.LongTensor],
+                unused_task_name: str="") -> torch.FloatTensor:
+        """ Run BERT to get hidden states.
+
+        Args:
+            sent: batch dictionary
+
+        Returns:
+            h: [batch_size, seq_len, d_emb]
+        """
+        assert "bert_wpm_pretokenized" in sent
+        # <int32> [batch_size, var_seq_len]
+        var_ids = sent["bert_wpm_pretokenized"]
+
+        # Model has fixed, learned positional component, so we must pass a
+        # block of exactly seq_len length.
```

Comment:
Since (I think) the model will always load positional embeddings for 512(?) positions, does `seq_len` need to be set to 512 when using a BERT model?

Reply:
I don't think so, < 512 should also work since BERT has an internal attention mask (https://github.com/huggingface/pytorch-pretrained-BERT/blob/master/pytorch_pretrained_bert/modeling.py#L620)

Reply:
:+1: